### PR TITLE
set up testthat infrastructure

### PR DESCRIPTION
**Pull request by reikookamoto**
Creating this PR to start a discussion on incorporating testthat correctly

- I reorganized the directory structure of `tests` so it plays well with the workflow described in the [R Packages book](https://r-pkgs.org/testing-basics.html)

- I first removed the old contents of `test-recode-with-table.R` into a file called `old-test.R` (to be deleted later)
- Then, I tried wrapping each test in `test_that()` and added it back into `test-recode-with-table.R`
- Now, if you call `devtools::test()`, you'll find that the 7 tests in `test-recode-with-table.R` 2 pass
- However, there are 2 tests in `old-test.R` that fail when you try compartementalizing them into separate `test_that()` calls
   - One involves a derived variable that depends on another variable, so I think this is related to the problem discussed in #37 - **can you confirm @yulric?** 
   - I haven't figured out why the other test is failing

tl;dr testthat infrastructure is now set up correctly, but 2 out of 9 previously written tests are failing

### CODE REVIEW

**Review by yulric — CHANGES_REQUESTED** [review]
Looks good, one potentially minor review comment. 

I fixed the issue with the failing tests, look at the latest commit I pushed and the message.

Regarding commits, we try to put in information about the changes made in a commit and why they were made. Take a look at the latest commit for more information. Lets try to follow that in the future.

**Review by yulric — APPROVED** [review]
